.github/workflows/ci_pr_tests.yml

```
@@ -0,0 +1,16 @@
+on: [push, pull_request]
```

To avoid duplicate runs:

```suggestion
on:
  push:
    branches:
      - main
  pull_request:
```